### PR TITLE
Presets for CAN mapping, OBDII and Analog channels

### LINKED ISSUE #985: Organize and store OBD-II mapping presets
Similar to https://github.com/autosportlabs/RaceCapture_App/issues/983, have the ability to select from a list of presets (presumably year / make /model vehicle) and auto populate a list of known PIDs, including known OEM specific PIDs.

Similar to trackmap management:
- Pre-load app distribution with a default collection of currently known mappings
- Ability to download updated preset mappings

### PR DESCRIPTION
**Pull request by brentpicasso**
This hooks up presets for the CAN mapping, OBDII and analog channel view. 

resolves #985 #983 #783 

also re-worked look of analog mapping view, and enabled the linear offset field in addition to multiplier so it looks like a linear formula - #491 #1322

Pane of map vs formula vs none updates based on the selected mapping type.

image

image

image

### CODE REVIEW

**Comment by ddimensia** [comment]
Just an initial comment on the screen shots above, could be done as a follow-up issue/PR.

It would be nice if all three modes of setting up the analog gauges kept the graph to the right so that it shows what the result would be.  Even though it seems obvious what raw 0-5v will be I think it's better for it to be consistent so users have a visual representation.  This would also allow the UI to have a consistent layout and not jump around.

Also, in the formula case it would be nice if you could put labels below the multiplier and offset text inputs, e.g. (multiplier), (offset).

File: autosportlabs/racecapture/presets/presetview.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
+import kivy
```
Comment (ddimensia) [inline autosportlabs/racecapture/presets/presetview.py]: Copyright header

File: autosportlabs/racecapture/views/configuration/rcp/canchannelsview.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from kivy.uix.popup import Popup
```
Comment (ddimensia) [inline autosportlabs/racecapture/views/configuration/rcp/canchannelsview.py]: Add Copyright declaration

**Comment by brentpicasso** [comment]
You're right, having the chart there all the time would make sense to illustrate the effects of the formula or mapping. I'll see what can be done in the short term.

File: autosportlabs/racecapture/presets/__init__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#
+# Race Capture App
+#
+# Copyright (C) 2014-2016 Autosport Labs
```
[the remark this answers is not in the document]
Comment (brentpicasso) [reply autosportlabs/racecapture/presets/__init__.py]: will sweep through the main codebase to handle that, in commit on the main branch.

**Comment by brentpicasso** [comment]
Reworked analog mapping view so graph is always visible: 

blob:https://autosportlabs.imgur.com/766d96cc-f0c0-40a7-899e-385fee947c3f

**Pull request merged**